internal/workload/podspec_updates_test.go

```diff
@@ -916,6 +915,73 @@ func TestPodTemplateAnnotations(t *testing.T) {
 
 }
 
+func TestTelemetryAddsTelemetryContainerPort(t *testing.T) {
+
+	var u = workload.NewUpdater("cloud-sql-proxy-operator/dev", workload.DefaultProxyImage)
+
+	// Create a pod
+	wl := podWorkload()
+	wl.Pod.Spec.Containers[0].Ports =
+		[]corev1.ContainerPort{{Name: "http", ContainerPort: 8080}}
+
+	// Create a AuthProxyWorkload that matches the deployment
+	csqls := []*cloudsqlapi.AuthProxyWorkload{
+		simpleAuthProxy("instance1", "project:server:db"),
+		simpleAuthProxy("instance2", "project:server2:db2"),
+		simpleAuthProxy("instance3", "project:server3:db3"),
+	}
+
+	// explicitly configure the telemetry http port for test consistency.
+	csqls[0].Spec.AuthProxyContainer = &cloudsqlapi.AuthProxyContainerSpec{
+		Telemetry: &cloudsqlapi.TelemetrySpec{
+			HTTPPort: ptr(workload.DefaultHealthCheckPort),
+		},
+	}
+	csqls[1].Spec.AuthProxyContainer = &cloudsqlapi.AuthProxyContainerSpec{
+		Telemetry: &cloudsqlapi.TelemetrySpec{
+			HTTPPort: ptr(workload.DefaultHealthCheckPort + 1),
+		},
+	}
+	csqls[2].Spec.AuthProxyContainer = &cloudsqlapi.AuthProxyContainerSpec{
+		Telemetry: &cloudsqlapi.TelemetrySpec{
+			HTTPPort: ptr(workload.DefaultHealthCheckPort + 2),
+		},
+	}
+
+	var wantPorts = map[string]int32{
```

Comment:
nit: let's move this down closest to where it's used.

Reply:
Done.